pkgs/top-level/stage.nix

```diff
@@ -75,6 +75,11 @@ let
       inherit (self) runtimeShell;
     };
 
+  pkgsLib = self: super: import ../pkgs-lib {
```

Comment:
An overlay is unnecessary because the same could be achieved by putting `formats` in `all-packages.nix`.

Categorizing functions separately from packages is beyond the scope of this change. It can be discussed in a separate issue/pr, although I doubt its usefulness.

Reply:
Doesn't have to be just functions though, can be arbitrary values, and other functions like `fetchgit` wouldn't fit in a pkgs-lib. I guess it won't be a problem to just put `formats` in `all-packages.nix`, but imo having a separation between "things needed to create packages" and "things that just happen to depend on pkgs" would make sense. Could still introduce that in the future though

Reply:
Not using an overlay anymore now

Reply:
All of `formats`, `fetchgit`, `dockerTools` depend on Nixpkgs, take arguments, produce derivations, are maintained and released in the same way. What makes `formats` so architecturally different that it needs to be treated differently?

> imo having a separation between "things needed to create packages" and "things that just happen to depend on pkgs" would make sense. Could still introduce that in the future though

If only one attribute is separate it doesn't make sense though. For it to make sense it must be applied to the whole of Nixpkgs and it must have practical benefits. The prior can be done in the future and I'm skeptical about the benefits, but would love to be proven wrong in that same future.

For now please follow the example of, say, `dockerTools`.

Reply:
Looks like my view of the PR was outdated (maybe caused by the current service disruption?) anyway, thanks for removing the overlay. Currently `build-support` has the role of `pkgs-lib` so you may want to move the files to `build-support/formats`. I don't really like that name either so I'm going to leave this change up to you to decide.

Reply:
The difference between e.g. `fetchgit` and `formats.json` is that `fetchgit` is needed for building many of the `pkgs.*` derivations, whereas `formats.json` doesn't have any use for any `pkgs.*`. Instead `formats` are mostly just useful for NixOS modules (really, the module system in general, so it wouldn't fit in `nixos/lib`)

This is also why I don't think `build-support` would make sense, because `formats.json` doesn't support any package building.